rfc/rfc-0000-blueprint-match-fields.md

```diff
@@ -0,0 +1,95 @@
+# Meta
+[meta]: #meta
+- Name: Add matchFields and matchLabels to top level of blueprints
+- Start Date: 2022-02-10
+- Author(s): @jwntrs
+- Status: Draft
+- RFC Pull Request:
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The [template switching RFC](https://github.com/vmware-tanzu/cartographer/pull/75) introduced template `options` that included a `matchFields` selector. We should include this same selector at the `blueprint` level, and we should move the current labels to a `matchLabels` selector.
+
+# Motivation
+[motivation]: #motivation
+
+- Allow entire `blueprints` to match an `owner` based on certain fields.
+
+- Ensure that the behaviour of all `selectors` is consistent.
+
+# What it is
+[what-it-is]: #what-it-is
+
+```yaml
+apiVersion: carto.run/v1alpha2                                 # <======== bump this
+kind: ClusterSupplyChain
+metadata:
+  name: supply-chain
+spec:
+  selector:
+    matchLabels:                                                # <=========== move existing labels under this heading
+      app.tanzu.vmware.com/workload-type: web
+    matchFields:                                                # <=========== add this
+      - { key: ".spec.image", operation: exists }               # <=========== 
```

Comment:
not sure if worth calling out or not, but for template switching we end up going without `.` in the beginning of the jsonpath (e.g., `spec.image` rather than `.spec.image`), which is actually consistent with other core k8s resources ala daemonset:

```yaml
nodeAffinity:
  requiredDuringSchedulingIgnoredDuringExecution:
    nodeSelectorTerms:
    - matchFields:
      - key: metadata.name
        operator: In
        values:
        - target-host-name
```

(see https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#scheduled-by-default-scheduler)

cartographer implementation for template switching: https://github.com/vmware-tanzu/cartographer/blob/3824091dd70ea53679d12ef380bd898cb1ec50d4/pkg/apis/v1alpha1/workload.go#L61-L69

Reply:
@cirocosta makes sense. FWIW I'm also proposing that we remove the workload prefix: https://github.com/vmware-tanzu/cartographer/pull/653